src/sdi-helpers.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "sdi-helpers.h"
+
+GAppInfo *sdi_get_desktop_file_from_snap(SnapdSnap *snap) {
+  GPtrArray *apps = snapd_snap_get_apps(snap);
+  if ((apps == NULL) || (apps->len == 0)) {
+    return NULL;
+  }
+
+  if (apps->len > 1) {
```

Comment:
Nit: You could have less indentation by returning early from the function is `apps->len < 1`.

Reply:
Good point. Done.